auto_dev/contracts/contract_scafolder.py

```diff
@@ -61,7 +62,7 @@ def generate_openaea_contract(self, contract: Contract):
             msg = "Failed to initialise agent lib."
             raise ValueError(msg)
 
-        with isolated_filesystem():
+        with rollback.on_exception(Path.cwd()):
```

Comment:
this is gonna be great,

Only issue i can foresee is that currently, we use sys.exit(1) such that we exit from the cli.

Id like to basically raise exceptions and then handle them in the cli code with a sys.exit

That makes the output from the tool look a lot nicer as then we dont have like the stack trace every time it throws if that makes sense.